lib/utils/compose.coffee

```diff
@@ -423,18 +452,20 @@ exports.deployProject = (
 		tagServiceImages(docker, images, serviceImages)
 		.tap (images) ->
 			logger.logDebug('Authorizing push...')
-			authorizePush(apiEndpoint, images[0].registry, _.map(images, 'repo'))
-			.then (token) ->
-				logger.logInfo('Pushing images to registry...')
-				pushAndUpdateServiceImages docker, token, images, (serviceImage) ->
-					logger.logDebug("Saving image #{serviceImage.is_stored_at__image_location}")
-					if skipLogUpload
-						delete serviceImage.build_log
-					releaseMod.updateImage(client, serviceImage.id, serviceImage)
-			.finally ->
-				logger.logDebug('Untagging images...')
-				Promise.map images, ({ localImage }) ->
-					localImage.remove()
+			getPreviousRepos(logger, appId)
+			.then (previousRepos) ->
+				authorizePush(apiEndpoint, images[0].registry, _.map(images, 'repo'), previousRepos)
+				.then (token) ->
+					logger.logInfo('Pushing images to registry...')
+					pushAndUpdateServiceImages docker, token, images, (serviceImage) ->
+						logger.logDebug("Saving image #{serviceImage.is_stored_at__image_location}")
+						if skipLogUpload
+							delete serviceImage.build_log
+						releaseMod.updateImage(client, serviceImage.id, serviceImage)
+				.finally ->
+					logger.logDebug('Untagging images...')
+					Promise.map images, ({ localImage }) ->
+						localImage.remove()
```

Comment:
This `.finally` block seems to have the purpose of undoing the earlier `tagServiceImages` call, "no matter what". It needs to be indented one level less (to the left) in order to cover the case of `getPreviousRepos` failing (rejected promise). Actually, I think the whole block from `.then (token)` to `.finally` could/should be indented one level less:
```coffeescript
			getPreviousRepos(logger, appId)
			.then (previousRepos) ->
				authorizePush(apiEndpoint, images[0].registry, _.map(images, 'repo'), previousRepos)
			.then (token) ->
				logger.logInfo('Pushing images to registry...')
				pushAndUpdateServiceImages docker, token, images, (serviceImage) ->
					logger.logDebug("Saving image #{serviceImage.is_stored_at__image_location}")
					if skipLogUpload
						delete serviceImage.build_log
					releaseMod.updateImage(client, serviceImage.id, serviceImage)
			.finally ->
				logger.logDebug('Untagging images...')
				Promise.map images, ({ localImage }) ->
					localImage.remove()
```

Reply:
Good catch, thanks!